aten/src/ATen/native/mps/operations/Pooling.mm

```diff
@@ -224,7 +224,7 @@ static void pool2d_template(const Tensor& input,
     runMPSGraph(mpsStream, cachedGraph->graph(), feeds, results);
 
     if (output_memory_format != suggested_memory_format) {
-      const_cast<Tensor&>(output) = output.to(suggested_memory_format);
+      output.copy_(output.to(suggested_memory_format));
```

Comment:
The `.to()` is not doing anything here since the current `output` layout is preserve during copy.

Reply:
True, and it clearly regresses `test_memory_format_nn_AdaptiveMaxPool2d_mps_float32`, so keeping old one here for a bit:
```shell
% python ../test/test_modules.py -v -k test_memory_format_nn_AdaptiveMaxPool2d_mps_float32
Fail to import hypothesis in common_utils, tests are not derandomized
test_memory_format_nn_AdaptiveMaxPool2d_mps_float32 (__main__.TestModuleMPS) ... FAIL

======================================================================
FAIL: test_memory_format_nn_AdaptiveMaxPool2d_mps_float32 (__main__.TestModuleMPS)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/malfet/git/pytorch/pytorch/torch/testing/_internal/common_utils.py", line 2354, in wrapper
    method(*args, **kwargs)
  File "/Users/malfet/git/pytorch/pytorch/torch/testing/_internal/common_device_type.py", line 415, in instantiated_test
    result = test(self, **param_kwargs)
  File "/Users/malfet/git/pytorch/pytorch/torch/testing/_internal/common_modules.py", line 118, in test_wrapper
    return test(*args, **kwargs)
  File "/Users/malfet/git/pytorch/pytorch/torch/testing/_internal/common_cuda.py", line 162, in wrapped
    return f(*args, **kwargs)
  File "/Users/malfet/git/pytorch/pytorch/build/../test/test_modules.py", line 698, in test_memory_format
    _check_out_mem_format(outputs, input_mem_format, module_mem_format)
  File "/Users/malfet/git/pytorch/pytorch/build/../test/test_modules.py", line 656, in _check_out_mem_format
    return self._traverse_obj(output, inner_check_out_mem_format)
  File "/Users/malfet/git/pytorch/pytorch/build/../test/test_modules.py", line 304, in _traverse_obj
    return func(obj)
  File "/Users/malfet/git/pytorch/pytorch/build/../test/test_modules.py", line 650, in inner_check_out_mem_format
    self.assertTrue(output.is_contiguous(memory_format=torch.channels_last))
AssertionError: False is not true
```